crates/l2/contracts/src/l1/CommonBridge.sol

```diff
@@ -200,7 +228,75 @@ contract CommonBridge is
         uint256 withdrawalBatchNumber,
         uint256 withdrawalLogIndex,
         bytes32[] calldata withdrawalProof
+    ) public {
+        _claimWithdrawal(
+            l2WithdrawalTxHash,
+            ETH_TOKEN,
+            ETH_TOKEN,
+            claimedAmount,
+            withdrawalBatchNumber,
+            withdrawalLogIndex,
+            withdrawalProof
+        );
+        (bool success, ) = payable(msg.sender).call{value: claimedAmount}("");
+        require(success, "CommonBridge: failed to send the claimed amount");
+    }
+
+    /// @inheritdoc ICommonBridge
+    function claimWithdrawalERC20(
+        bytes32 l2WithdrawalTxHash,
+        address tokenL1,
+        address tokenL2,
+        uint256 claimedAmount,
+        uint256 withdrawalBatchNumber,
+        uint256 withdrawalLogIndex,
+        bytes32[] calldata withdrawalProof
     ) public nonReentrant {
+        _claimWithdrawal(
+            l2WithdrawalTxHash,
+            tokenL1,
+            tokenL2,
+            claimedAmount,
+            withdrawalBatchNumber,
+            withdrawalLogIndex,
+            withdrawalProof
+        );
+        require(tokenL1 != ETH_TOKEN, "CommonBridge: attempted to withdraw ETH as if it were ERC20, use claimWithdrawal()");
+        IERC20(tokenL1).safeTransfer(msg.sender, claimedAmount);
+    }
+
+    function _claimWithdrawal(
```

Comment:
Let's find better names for this functions, we have `claimWithdrawal`, `_claimWithdrawal`, `claimWithdrawalERC20`, `_claimWithdrawalProof` and `_verifyWithdrawProof` (if I'm not missing anyone else). It's hard to understand what does each one do